src/scripts/app.ts

```diff
@@ -106,35 +112,77 @@ async function addItems(chosedValue: number, imageArray: Basic[]) {
 		showToast('error', 'Cannot load images');
 	}
 
-	return createFindContainer(selectedImages);
+	createToFindItem();
 }
 
-async function createFindContainer(selectedImages: Array<object>) {
-	const findImageSection = imageArray.slice(0, 1);
-	const randomImg = Math.floor(Math.random() * selectedImages.length);
-	console.log(imgItemsArray);
-
-	findImageSection.forEach((imgItemsArray: Basic) => {
-		const findContainer: HTMLDivElement =
-			document.querySelector('.js-find-container')!;
-		const findTitle: HTMLElement = document.querySelector('.js-find-title')!;
-		const imgToFind = document.createElement('img');
-		imgToFind.id = imgItemsArray.id[randomImg];
-		imgToFind.src = imgItemsArray.urls.small;
-		imgToFind.classList.add('image-to-find');
-		findContainer.append(imgToFind, findTitle);
-	});
+function createToFindItem() {
+	const randomImg = Math.floor(Math.random() * imgItemsArray.length);
+	const originalImage = imgItemsArray[randomImg];
+
+	const findContainer: HTMLDivElement =
+		document.querySelector('.js-find-container')!;
+	const findTitle: HTMLElement = document.querySelector('.js-find-title')!;
+	const imgToFind = document.createElement('img');
+
+	imgToFind.id = originalImage.id;
+	imgToFind.src = originalImage.src;
+	imgToFind.classList.add('image-to-find');
+	findContainer.append(imgToFind, findTitle);
 }
 
-const checkMatch = (e: Event) => {
-	console.log(e.target);
+const checkElements = (event: Event) => {
+	const clickedImg = event.target as HTMLElement;
+	const imgFind = document.querySelector('.image-to-find') as HTMLImageElement;
+
+	const clickedSrcValue = clickedImg.getAttribute('src');
+	const srcToFindValue = imgFind.getAttribute('src');
+	const clickedIdValue = clickedImg.getAttribute('id');
+	const idToFind = imgFind.getAttribute('id');
+
+	if (attemptCounter >= maxAttempts) {
+		showToast('error', 'Brak dalszych prób');
+		return;
+	}
+
+	if (clickedImg.classList.contains('js-game-img')) {
+		attemptCounter++;
+		console.log('clicked od the image', clickedImg);
+		checkMatch(clickedSrcValue, srcToFindValue, idToFind, clickedIdValue);
+	} else {
+		showToast('warning', 'You must choose an image');
+	}
+};
+
+const checkMatch = (
+	clickedSrcValue: string | null,
+	srcToFindValue: string | null | undefined,
+	idToFind: string | null,
+	clickedIdValue: string | null
+) => {
+	if (
+		clickedSrcValue !== null &&
+		srcToFindValue !== null &&
+		clickedIdValue !== null
```

Comment:
Zeby tego uniknac, w poprzedniej funkcji ktora tą wywołuje dodalbym ifa ktory to sprawdzi clickedSrcValue && srcToFindValue && idToFind && clickedIdValue, wtedy tutaj mamy pewnosc co nam przyjdzie